client/src/components/googleMap/googleMap.jsx

```diff
@@ -0,0 +1,188 @@
+		this.showMarkers(null);
+	}
+
+	//Removes all markers from map and clears state
+	deleteMarkers() {
+		console.log("running delete markers");
+		this.hideMarkers();
+		this.setState({ markers: [] });
+	}
+
+	//Updates the marker location on the map and removes the previous markers
+	updateLocationMarker(location, map) {
+		console.log("running update location markers");
+		this.deleteMarkers();
+		this.addMarker(location, map);
+	}
+
+	createMarkers(map) {
+		console.log("running createMarkers");
+		const sightings = this.props.sightings;
+		if (!sightings || sightings.length < 0) return null;
+
+		sightings.forEach(element => {
+			const defaultLocation = { lat: 45.63177710291909, lng: -79.71027154237892 };	
+			
+			let location;
+			try {
+				location = JSON.parse(element.location);
+			}
+			catch(err) {
+				location = defaultLocation;
+			}
+
+			const marker = new google.maps.Marker({
+				position: location,
+				map: map
+			});
+
+			//Some test DB entries don't have users assigned to them
+			let popUpInfo;
+			if (element.spottedBy) {
+				popUpInfo = '<div><h1>' + element.animal + '</h1><h3>' + element.spottedBy + '</h3></div>';
+			} else {
+				popUpInfo = '<div><h1>' + element.animal + '</h1><h3>Mystery User</h3></div>';
+			}
+			
+			const infoWindow = new google.maps.InfoWindow({
+				content: popUpInfo
+			});
+
+			marker.addListener('click', () => {
+				infoWindow.open(map, marker)
+			});
+		});
+	};
+
+	createDisplayMap() {
+		loader.load()
+		.then(() => {
+			//Create map
+			const map = new google.maps.Map(this.mapRef.current, {
+				center: { lat: 45.63177710291909, lng: -79.71027154237892 },
+				zoom: 15,
+			});
+
+			//console.log("done creating map");
+
+			//Add markers for all the sightings passed to the map
+			this.createMarkers(map);
+		});
+	};
+
+	createSightingMap() {
+		loader.load()
+		.then(() => {
+			//Create map
+			console.log("Creating sighting Map");
+			let createMap = new google.maps.Map(this.mapRef.current, {
+				center: { lat: 45.63177710291909, lng: -79.71027154237892 },
+				zoom: 15,
+			});
+
+			//Add a marker whenever you click on the map
+			const marker = new google.maps.Marker({
+				position: this.props.location,
+			});
+			marker.setMap(createMap);
+
+			/*
+			Need a way to pass createMap into the event listener function
+			Right now location (2nd parameter to udateLocation Marker) is always undefined
+			*/
+
+			google.maps.event.addListener(createMap, 'click', (event) => {
+/* 				this.setState({ location: event.latLng });
+
+				this.addEventListener(event, this)
+				this.updateLocationMarker(event.latLng, event.target); */
+
+				//Pass location back to previous state if requested
+				if (this.props.onClick) {
+					this.props.onClick(event.latLng)
+				}
+				console.log("Done with event Listener");
+			});
+			//google.maps.event.addListener(createMap, 'click', this.addEventListener(event, createMap));
+
+		});
+	};
+
+	addEventListener(event, map) {
```

Comment:
This isn't used anywhere?

Reply:
This was part of me trying to get the map to load a marker without refreshing. With how it's laid out right now, it is not required.